sensor/kubernetes/fake/flows.go

```diff
@@ -2,18 +2,99 @@ package fake
 	"github.com/stackrox/rox/sensor/common/networkflow/manager"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// OriginatorCache stores originators by endpoint key (IP:port) to provide consistent
+// process-to-endpoint mappings with configurable reuse probability.
+// This improves test realism by simulating how processes typically bind to consistent endpoints.
+//
+// This cache implements probabilistic reuse to simulate:
+//   - High probability case: Single process consistently binds to the endpoint; a new process can bind
+//     to the same endpoint only after sending a close message.
+//   - Low probability case: New process takes over the endpoint without sending a close message
+//     (port is shared by multiple processes).
+type OriginatorCache struct {
+	cache map[string]*storage.NetworkProcessUniqueKey
+	lock  sync.RWMutex
+}
+
+// NewOriginatorCache creates a new cache for storing endpoint-to-originator mappings.
+func NewOriginatorCache() *OriginatorCache {
+	return &OriginatorCache{
+		cache: make(map[string]*storage.NetworkProcessUniqueKey),
+	}
+}
+
+// GetOrSetOriginator retrieves a cached originator for the endpoint or generates a new one.
+// The portSharingProbability parameter controls the probability of reusing an existing open endpoint by
+// a different process versus generating a new endpoint for the process (range: 0.0-1.0).
+//
+// This caching mechanism improves test realism by ensuring that network endpoints
+// typically close the endpoint before changing the originator (e.g., 95% default), while still allowing
+// for on-the-fly process changes (e.g., 5% for multiple processes reusing the same port in parallel).
+//
+// Note that generating multiple 'open' endpoints with the same IP and Port but different originators
+// without a 'close' message in between is not a realistic scenario and occurs very rarely in production
+// (practically only when a process deliberately abuses the same IP and Port in parallel to a different process).
+//
+// Setting the portSharingProbability to an unrealistic value (anything higher than 0.05) would cause additional memory-pressure
+// in Sensor enrichment pipeline because the deduping key for processes contains the IP, Port and the originator.
+// Note that if Sensor sees an open endpoint for <container1, 1.1.1.1:80, nginx> and then another open endpoint for
+// <container1, 1.1.1.1:80, apache2>, then Sensor will keep the nginx-entry forever, as there was no 'close' message in between.
+//
+// The probability logic is explicit and configurable for differ{ent testing scenarios.
+func (oc *OriginatorCache) GetOrSetOriginator(endpointKey string, containerID string, openPortReuseProbability float64, processPool *ProcessPool) *storage.NetworkProcessUniqueKey {
+	// Ensure that the probability is between 0.0 and 1.0.
+	prob := math.Min(1.0, math.Max(0.0, openPortReuseProbability))
+	if openPortReuseProbability < 0.0 || openPortReuseProbability > 1.0 {
```

Comment:
Maybe this check can be done just once when the configuration is parsed. Then you can also simplify the line above this one.

Reply:
The function name `validateWorkload` implies that there are many checks in it. I don't think that has to be done now. Maybe a comment that more checks should be added to it in the future is good enough for now.

Reply:
[nit] I was thinking that `openPortReuseProbability` could also be set to 0 or 1.0 when the validation is done, and then instead of using `prob` you could just use `openPortReuseProbability`.

Reply:
Oh, so like change it directly in the `workload` object? Yeah, I can do that